lib/semian/adapter.rb

```diff
@@ -53,7 +59,11 @@ def semian_options
       return @semian_options if defined? @semian_options
 
       options = raw_semian_options
-      @semian_options = options && options.map { |k, v| [k.to_sym, v] }.to_h
+
+      symbolized_options = options && options.map { |k, v| [k.to_sym, v] }.to_h
+      symbolized_options.tap do
+        @semian_options = symbolized_options if (symbolized_options || {}).fetch(:deterministic, true)
```

Comment:
It looks a bit complex line for me.

I think `symbolized_options` could be `nil`, only when `options` is `nil`.
In other cases `symbolized_options` is a Hash.

Also I have doubts of using `tap` in this section of changes. 
Can you help me to understand, why `tap` is required?

Do you know if it is possible to avoid this block

Reply:
Agreed, it does look overly complex.

The reasoning behind `tap` was to get an easy way to always return `symbolized_options` (regardless of the value for `:deterministic`). The alternative version I started with (and ended up refactoring to the `tap` version) looked like this:

```ruby
symbolized_options = options && options.transform_keys(&:to_sym)
@semian_options = symbolized_options if (symbolized_options || {}).fetch(:deterministic, true)
symbolized_options
```

The `symbolized_options || {}` piece was for cases where `raw_semian_options` returns `false` or `nil`.

With all that said, would you prefer a `tap`-less, and without the `|| {}` thingy, like this: 

```
symbolized_options = options && options.transform_keys(&:to_sym)
@semian_options = symbolized_options if !symbolized_options || symbolized_options.fetch(:deterministic, true)
symbolized_options
```